bundles/org.openhab.binding.epsonprojector/src/main/java/org/openhab/binding/epsonprojector/internal/discovery/EpsonProjectorDiscoveryParticipant.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.epsonprojector.internal.discovery;
+
+import static org.openhab.binding.epsonprojector.internal.EpsonProjectorBindingConstants.*;
+
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.discovery.DiscoveryResult;
+import org.openhab.core.config.discovery.DiscoveryResultBuilder;
+import org.openhab.core.config.discovery.sddp.SddpDevice;
+import org.openhab.core.config.discovery.sddp.SddpDiscoveryParticipant;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * Discovery Service for Epson Projectors that support SDDP.
+ *
+ * @author Michael Lobstein - Initial contribution
+ *
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class EpsonProjectorDiscoveryParticipant implements SddpDiscoveryParticipant {
+    private final Logger logger = LoggerFactory.getLogger(EpsonProjectorDiscoveryParticipant.class);
+
+    private static final String EPSON = "EPSON";
+    private static final String TYPE_PROJECTOR = "PROJECTOR";
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(SddpDevice device) {
+        final ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            final Map<String, Object> properties = new HashMap<>(3);
+            final String label = device.manufacturer + " " + device.model;
+
+            properties.put(Thing.PROPERTY_MAC_ADDRESS, uid.getId());
+            properties.put(THING_PROPERTY_HOST, device.ipAddress);
+            properties.put(THING_PROPERTY_PORT, DEFAULT_PORT);
```

Comment:
Why not using `device.port` ?

Reply:
The port from the SDDP packet does not help since it is the multicast port. For Epson with built-in Ethernet, they listen on TCP 3629 for the control connection. This thing also supports serial-over-IP and in that scenario the user can specify their desired port. For this use case when creating a discovery result based on an SDDP packet, we always want the port to be 3629.